transcom/mymove
# MAIN - B-18679 - New QAE/CSR table columns

## [B-18679](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A905412&RoomContext=TeamRoom%3A855664&concept=TeamRoom)

## [INT PR](https://github.com/transcom/mymove/pull/12172)

## Summary
For MAIN

This story adds Origin GBLOC, Scheduled Pickup Date, Destination GBLOC, and Scheduled Delivery Date columns to the QAE/CSR user's search results table.

Is there anything you would like reviewers to give additional scrutiny?
Since each shipment in a move can have its own scheduled dates this uses the date from the first shipment.

React-table recently switched to v8 and no longer hosts docs for v7 (what we currently use). I found [React-Table v7 Docs here](https://react-table-v7-docs.netlify.app/docs)

<img width="1437" alt="Screenshot 2024-03-12 at 12 54 09 PM" src="https://github.com/transcom/mymove/assets/158500142/71feecf1-a69b-41bd-86ae-620641681051">

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Ensure there are several moves available to view as a QAE/CSR user. At least one PPM and at least one HHG with scheduled pickup and delivery dates.
2. Login as a QAE/CSR user
3. Search for moves via any of the three search types
4. Ensure all four new columns are shown and display accurate data. The date columns should be empty unless there is a scheduled pickup or delivery date.
5. Ensure sorting and filtering function correctly on all columns that allow it.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
